src/test/java/htsjdk/tribble/index/IndexFactoryTest.java

```diff
@@ -112,4 +117,45 @@ public void testCreateTabixIndexOnBlockCompressed() {
                     "Tabix indexed (bgzipped) VCF does not contain sequence: " + samSequenceRecord.getSequenceName());
         }
     }
+
+    @Test
+    public void testWriteIndex() throws Exception {
```

Comment:
Since this test is in indexFactory.writeIndex() it should probably be testing `indexFactory.writeIndex()`...  

Test for `write` on the index files themselves should live in TabixIndexTest and LinearIndexTest respectively. 

Either way (or both might be the best at the risk of redundantly testing things), `TabixIndexTest.readWriteTest` provides a good model for how the test might be made.  Instead of repeating the code for the different cases you can write a test function that takes a dataprovider with several inputs.  It might be a pain though for linear indexes since I don't see a read or equals method on them.